department-of-veterans-affairs/caseflow
# CASEFLOW-138 Propagate task instructions when task is bulk assigned

Resolves [CASEFLOW-138](https://vajira.max.gov/browse/CASEFLOW-138)

### Description
Propagate task instructions when task is bulk assigned


### Acceptance Criteria
- [ ] Task Instructions Remain When Task is Bulk Assigned

### Testing Plan
(See Testing Plan in #15125. Borrowed from Testing Plan in #14523, except include instructions and check for instructions in newly assigned tasks.)

1. Make Jolly Postman an admin
   ```ruby
   OrganizationsUser.make_user_admin(User.find_by(css_id: "JOLLY_POSTMAN"), MailTeam.singleton)
   ```
1. Log in as jolly postman and go to the mail team queue
1. Go to Case Details for one of the Unassigned cases and grab the UUID from the URL
1. Add instructions to the active task of the selected appeal:
```ruby
t=Task.find(NNN)
t.update(instructions: ["do good"])
```
1. Refresh Case Details and click View Instructions to see the instruction.
1. Back in the queue view, click "Assign tasks"
1. Assign some tasks to a user. Set the filters so that the case you modified will be bulk assigned.
1. Confirm tasks were assigned to the user. See the Assigned tab.
1. Open the appeal you modified to ensure the instructions propagated to the new user-task. Check the task instructions in the task tree:
```ruby
t.children.active.first.instructions
```

Perform the same Testing Plan in the `master` branch and check that `instructions` are not propagated.

### User Facing Changes

 BEFORE|AFTER
 ---|---
![image](https://user-images.githubusercontent.com/55255674/107680554-9c9c8e80-6c63-11eb-9b2b-2641dcac359f.png) | ![image](https://user-images.githubusercontent.com/55255674/107680242-36177080-6c63-11eb-869c-78c0cfcc5f81.png)
